crates/sui-core/src/safe_client.rs

```diff
@@ -429,17 +423,30 @@ where
         &self,
         request: CheckpointRequest,
     ) -> Result<CheckpointResponse, SuiError> {
+        let detail = request.detail;
         let req_type = request.request_type.clone();
 
         let resp = self.authority_client.handle_checkpoint(request).await?;
 
-        match req_type {
+        // Verify signatures
+        resp.verify(&self.committee)?;
+
+        if let (CheckpointRequestType::PastCheckpoint(_), true, None) =
+            (&req_type, detail, &resp.detail)
+        {
+            // detail was requested but not returned. Note for
+            // CheckpointRequestType::LatestCheckpointProposal, detail may not be available and so
+            // it may be present or absent.
+            return Err(SuiError::ByzantineAuthoritySuspicion {
```

Comment:
This seems still off. What if the validator just doesn't have this checkpoint?

Reply:
you're right, i need to allow for the case where the authority returns None

Reply:
fixed now.